pkgs/ok_http/lib/src/ok_http_client.dart

```diff
@@ -126,6 +126,9 @@ class OkHttpClient extends BaseClient {
               request: request,
               contentLength: contentLength,
             ));
+
+            // Close and remove all idle connections from the resource pool.
+            _client.connectionPool().evictAll();
```

Comment:
I read your explanation but I still don't understand why this is necessary. Since the socket is closed by the server after receiving the headers, how can it be reused?

https://github.com/dart-lang/http/blob/9f022d26364fb09748cbca25834b23fdc26baf05/pkgs/http_client_conformance_tests/lib/src/request_cookies_server.dart#L48

Reply:
The problem here is that even if the socket is closed, the server keeps listening to the request
```dart
request.listen((line)) {
```
and even when you encounter an empty line, it only adds the `cookies` list to the channel sink.

This seemed very peculiar to me, but I confirmed that it was indeed the problem by doing:
If you create a StreamSubscription and cancel it when an empty line is encountered.
```dart
late StreamSubscription reqSubscription;
reqSubscription = request.listen((line) {
  if (line.toLowerCase().startsWith('cookie:')) {
    cookies.add(line);
  }

  if (line.isEmpty) {
    // A blank line indicates the end of the headers.
    channel.sink.add(cookies);
    reqSubscription.cancel(); // <- here
  }
});
```
then all works well, without the need of explicitly removing idle connections.

Reply:
This is what I used for reference: https://stackoverflow.com/questions/31183730/okhttp-asynchronous-request-doesnt-stop-immediately-after-onresponse

Reply:
Hmmm... the test seems racy. Does this formulation work:

```dart
await for (final line in request) {
        if (line.toLowerCase().startsWith('cookie:')) {
          cookies.add(line);
        }

        if (line.isEmpty) {
          // A blank line indicates the end of the headers.
          channel.sink.add(cookies);
          break;
        }
}

socket.writeAll(...)
await socket.close()
```

Reply:
Yes, it does!

Reply:
Do you suggest changing the test case instead? I didn't go that route because `CronetClient` seems to pass both cases without a problem.

Reply:
> Do you suggest changing the test case instead? I didn't go that route because `CronetClient` seems to pass both cases without a problem.

Yeah, I think that the test is incorrect and it is a coincidence that it works ;-)

Reply:
Alright, then it's good to close this PR. We would still need `okhttp3.ConnectionPool` and `evictAll()` for properly closing the client if I'm not wrong.

Reply:
> Alright, then it's good to close this PR. We would still need `okhttp3.ConnectionPool` and `evictAll()` for properly closing the client if I'm not wrong.

You probably want to empty the connection pool when in response to `Client.close()`. From the [docs](https://square.github.io/okhttp/5.x/okhttp/okhttp3/-ok-http-client/index.html#:~:text=Shutdown%20Isn't%20Necessary,executor%20service%20with%20shutdown())

Reply:
Yeah, that's exactly what I was going to follow.
1. call shutdown on `ExecutorService`
2. empty the connection pool
3. release client object (JNI)